Content.Client/TapeRecorder/Ui/TapeRecorderMenu.xaml.cs

```diff
@@ -0,0 +1,154 @@
+    private readonly TapeRecorderBoundUserInterface _owner;
+
+    private TapeRecorderMode _mode = TapeRecorderMode.Empty;
+    private bool _active = false;
+    private float _rewindSpeed = 3f;
+    private bool _onCooldown = false;
+
+    public TapeRecorderMenu(TapeRecorderBoundUserInterface owner)
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        _owner = owner;
+
+        PlayButton.Group = _controlButtons;
+        StopButton.Group = _controlButtons;
+
+        PlayButton.OnPressed += _ => _owner.ToggleSwitch();
+        StopButton.OnPressed += _ => _owner.ToggleSwitch();
+
+        RewindButton.Group = _modeButtons;
+        RecordButton.Group = _modeButtons;
+        PlaybackButton.Group = _modeButtons;
+
+        RewindButton.OnPressed += _ => _owner.ChangeMode(TapeRecorderMode.Rewinding);
+        RecordButton.OnPressed += _ => _owner.ChangeMode(TapeRecorderMode.Recording);
+        PlaybackButton.OnPressed += _ => _owner.ChangeMode(TapeRecorderMode.Playing);
+
+        PrintButton.OnPressed += _ => _owner.PrintTranscript();
+    }
+
+    private void SetSlider(float maxTime, float currentTime)
+    {
+        PlaybackSlider.Disabled = true;
+        PlaybackSlider.MaxValue = maxTime;
+        PlaybackSlider.Value = currentTime;
+    }
+
+    public void UpdatePrint(bool disabled)
+    {
+        if (disabled)
+        {
+            PrintButton.Disabled = true;
+            _onCooldown = true;
+        }
+        else
+        {
+            PrintButton.Disabled = false;
+            _onCooldown = false;
+        }
+    }
+
+    public void UpdateState(TapeRecorderState state)
+    {
+        _rewindSpeed = state.RewindSpeed;
+        _active = state.Active;
+        _mode = state.Mode;
+
+        CassetteLabel.Text = (state.HasCasette)
+            ? Loc.GetString("tape-recorder-menu-cassette-label", ("cassetteName", state.CassetteName))
+            : Loc.GetString("tape-recorder-menu-no-cassette-label");
+
+        // Checking mode and enabling button to make it glow
+        switch (state.Mode)
+        {
+            case TapeRecorderMode.Recording:
+                {
+                    RecordButton.Pressed = true;
+                    break;
+                }
+            case TapeRecorderMode.Rewinding:
+                {
+                    RewindButton.Pressed = true;
+                    break;
+                }
+            case TapeRecorderMode.Playing:
+                {
+                    PlaybackButton.Pressed = true;
+                    break;
+                }
+            default:
+                break;
+        }
+
+        // Checking all buttons to now overdisable them in future
+        if ((!state.HasCasette || state.Active || state.CurrentTime >= state.MaxTime) && state.Mode != TapeRecorderMode.Playing)
+            PlaybackButton.Disabled = true;
+        else
+            PlaybackButton.Disabled = false;
```

Comment:
This should just be `PlaybackButton.Disabled = (condition)`, same for the others.